components/tabs/tab-mixin.js

```diff
@@ -0,0 +1,193 @@
+			display: inline-block;
+			max-width: 200px;
+			outline: none;
+			position: relative;
+			vertical-align: middle;
+		}
+		.d2l-tab-text {
+			margin: 0.5rem;
+			overflow: hidden;
+			padding: 0.1rem;
+			text-overflow: ellipsis;
+			white-space: nowrap;
+		}
+		:host(:first-child) .d2l-tab-text {
+			margin-inline-end: 0.6rem;
+			margin-inline-start: 0;
+		}
+		.d2l-tab-selected-indicator {
+			border-top: 4px solid var(--d2l-color-celestine);
+			border-top-left-radius: 4px;
+			border-top-right-radius: 4px;
+			bottom: 0;
+			display: none;
+			margin: 1px 0.6rem 0 0.6rem;
+			position: absolute;
+			transition: box-shadow 0.2s;
+			width: calc(100% - 1.2rem);
+		}
+		:host(:first-child) .d2l-tab-selected-indicator {
+			margin-inline-end: 0.6rem;
+			margin-inline-start: 0;
+			width: calc(100% - 0.6rem);
+		}
+		:host(:${unsafeCSS(getFocusPseudoClass())}) > .d2l-tab-text {
+			border-radius: 0.3rem;
+			box-shadow: 0 0 0 2px var(--d2l-color-celestine);
+			color: var(--d2l-color-celestine);
+		}
+		:host([aria-selected="true"]:focus) {
+			text-decoration: none;
+		}
+		:host(:hover) {
+			color: var(--d2l-color-celestine);
+			cursor: pointer;
+		}
+		:host([aria-selected="true"]:hover) {
+			color: inherit;
+			cursor: default;
+		}
+		:host([aria-selected="true"]) .d2l-tab-selected-indicator {
+			display: block;
+		}
+    `;
+
+	constructor() {
+		super();
+		this.ariaSelected = false;
+		this.role = 'tab';
+		this.selected = false;
+		this.tabIndex = -1;
+
+		this.#handleClickBound = this.#handleClick.bind(this);
+		this.#handleKeydownBound = this.#handleKeydown.bind(this);
+		this.#handleKeyupBound = this.#handleKeyup.bind(this);
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this.#addEventHandlers();
+
+		if (!this.#resizeObserver) {
+			this.#resizeObserver = new ResizeObserver(() => {
+				this.#handleResize();
+			});
+			this.#resizeObserver.observe(this);
+		}
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this.#removeEventHandlers();
+
+		if (this.#resizeObserver) {
+			this.#resizeObserver.disconnect();
+			this.#resizeObserver = null;
+		}
+	}
+
+	render() {
+		const overrideSkeletonText = this.skeleton && (!this.text || this.text.length === 0);
+		const contentClasses = {
+			'd2l-tab-text': true,
+			'd2l-skeletize': true,
+			'd2l-tab-text-skeletize-override': overrideSkeletonText
+		};
+
+		return html`
+			<div class="${classMap(contentClasses)}">
+				${this.renderContent}
+			</div>
+			<div class="d2l-tab-selected-indicator d2l-skeletize-container"></div>
+		`;
+	}
+
+	update(changedProperties) {
+		super.update(changedProperties);
+		changedProperties.forEach((oldVal, prop) => {
+			if (prop === 'selected') {
+				this.ariaSelected = this.selected;
+				if (this.selected === 'true') {
+					this.dispatchEvent(new CustomEvent(
+						'd2l-tab-selected', { bubbles: true, composed: true }
+					));
+				}
+			} else if (prop === 'text') {
+				this.title = this.text;
```

Comment:
We don't have the `text` reactive property defined anymore, but we're still intending to set the `title` here. 

Subclasses might have more complex text arrangements. For instance, text like we currently have in addition to a badge count or something. So maybe it makes sense for the subclass to handle this.

Reply:
Agreed that subclass managing text + title makes sense. Particularly for SR/AT